db/document.go

```diff
@@ -1129,19 +1179,27 @@ func (doc *Document) MarshalJSON() (data []byte, err error) {
 // (unmarshalLevel) specifies how much of the provided document/xattr needs to be initially unmarshalled.  If
 // unmarshalLevel is anything less than the full document + metadata, the raw data is retained for subsequent
 // lazy unmarshalling as needed.
-func (doc *Document) UnmarshalWithXattr(ctx context.Context, data []byte, xdata []byte, unmarshalLevel DocumentUnmarshalLevel) error {
+func (doc *Document) UnmarshalWithXattr(ctx context.Context, data, syncXattrData, hlvXattrData []byte, unmarshalLevel DocumentUnmarshalLevel) error {
 	if doc.ID == "" {
 		base.WarnfCtx(ctx, "Attempted to unmarshal document without ID set")
 		return errors.New("Document was unmarshalled without ID set")
 	}
 
 	switch unmarshalLevel {
 	case DocUnmarshalAll, DocUnmarshalSync:
```

Comment:
There's probably some ambiguity here about whether 'sync metadata' refers to both _sync and _vv xattrs (as it's the combination that makes up SGW's metadata).  I've updated the comments to just say 'metadata' to avoid confusion.